site/src/sphinx/advanced-retry.rst

```diff
@@ -1,8 +1,276 @@
+
+    import com.linecorp.armeria.client.ResponseTimeoutException;
+    import com.linecorp.armeria.common.HttpStatusClass;
+
+    new RetryStrategy<HttpRequest, HttpResponse>() {
+        final Backoff backoffOnServerErrorOrTimeout = RetryStrategy.defaultBackoff;
+        final Backoff backoffOnConflict = Backoff.fixed(100);
+
+        @Override
+        public CompletableFuture<Optional<Backoff>> shouldRetry(HttpRequest request,
+                                                                HttpResponse response) {
+            return response.aggregate().handle((result, cause) -> {
+                if (cause != null) {
+                    if (cause instanceof ResponseTimeoutException) {
+                        return Optional.of(backoffOnServerErrorOrTimeout);
+                    }
+                } else if (result.headers().status().codeClass() == HttpStatusClass.SERVER_ERROR) {
+                    return Optional.of(backoffOnServerErrorOrTimeout);
+                } else if (result.headers().status() == HttpStatus.CONFLICT) {
+                    return Optional.of(backoffOnConflict);
+                }
+                return Optional.empty();
+            });
+        }
+    };
+
+``Backoff``
+-----------
+
+You can use a Backoff_ to determine the delay between attempts. Armeria provides Backoff_ implementations which
+produce the following delays out of the box:
+
+- Fixed delay, created with ``Backoff.fixed()``
+- Random delay, created with ``Backoff.random()``
+- Exponential delay which is multiplied on each attempt, created with ``Backoff.exponential()``
+
+Armeria provides ``RetryStrategy.defaultBackoff`` that you might use as default. It is exactly same with:
+
+.. code-block:: java
+
+    Backoff.exponential(minDelayMillis /* 200 */, maxDelayMillis /* 10000 */, multiplier /* 2.0 */)
+           .withJitter(jitterRate /* 0.2 */)
+           .withMaxAttempts(maxAttempts /* 10 */);
+
+The delay starts from ``minDelayMillis`` until it reaches ``maxDelayMillis`` multiplying with multiplier.
+Please note that the ``.withJitter()`` will add jitter value to the calculated delay. This will attempts
+no more than ``maxAttempts``.
+
+For more information, please refer to the API documentation of the `com.linecorp.armeria.client.retry`_ package.
+
+``totalMaxAttempts`` vs per-Backoff ``maxAttempts``
+---------------------------------------------------
+
+If you create a Backoff_ using ``.withMaxAttempts(maxAttempts)`` in a RetryStrategy_, the RetryingClient_
+which uses the RetryStrategy_ will stop retrying when the number of attempts passed ``maxAttempts``.
+However, if you have more than one Backoff_ and return one after the other continuously, it will keep retrying
+over and over again because the counter that RetryingClient_ internally tracks is initialized every time the
+different Backoff_ is returned. If you want to limit the number of whole retry session,
+you need to use ``totalMaxAttempts`` when you create the RetryingClient_.
```

Comment:
- If you want to limit ... you need to use ... -> To limit the number of attempts in a whole retry session, RetryingClient_ limits the maximum number of total attempts to 10. You can change this value by specifying maxTotalAttempts when you build a RetryingClient_:: ...